text/0000-project-error-handling.md

```diff
@@ -43,6 +43,13 @@ Here is a tenative starting point, subject to change:
 - What is the consensus on handling `dyn Error`s? Should it be encouraged or discouraged? Should we look into making `Box<dyn Error...>` implement `Error`?
 
 
+### Identify pain points in error handling today
+
+- Backtrace capture is expensive, but without one it can be difficult to pinpoint the origin of errors
+- unwrap on errors without first converting to a reporting type will often discard relevant information
+- errors printing from main have to assume a prefixed `Error: `, sub par control of output format when printing during termination.
```

Comment:
```suggestion
- errors printed from `main` have to assume a prefixed `Error: `, sub-par control of output format when printing during termination.
```

Reply:
I don't know what "have to assume a prefixed `Error: `" means.

Reply:
I'm referring to this default in `std`'s impl of `Termination` for `Result` https://doc.rust-lang.org/stable/src/std/process.rs.html#1690-1696, right now there's no way to signal to `Debug` types that they're being formatted as part of `Termination` or not. @dtolnay has suggested adding an `is_termination` flag to `Debug`. I'm also interested in the possibility of a 3rd `std::fmt` trait called `std::fmt::Report` to handle this, leveraging specialization.